tests/sentry/utils/test_snuba.py

```diff
@@ -607,3 +611,44 @@ def test_aggregations(self):
 
         query = {"aggregations": [["uniq", "transaction.name", "uniq_transaction"]]}
         assert detect_dataset(query) == Dataset.Transactions
+
+
+class PrepareQueryParamsTest(TestCase):
+    def test_events_dataset_with_project_id(self):
+        query_params = SnubaQueryParams(
+            dataset=Dataset.Events, filter_keys={"project_id": [self.project.id]}
+        )
+
+        kwargs, _, _ = _prepare_query_params(query_params)
+        assert "project" in kwargs
+        assert kwargs["project"] == [self.project.id]
```

Comment:
Aren't these assertions redundant? Line 624 will `KeyError` unless 623 is also true. (Same several times below.)